tests/manage/z_cluster/cluster_expansion/test_add_non_ocs_workers.py

```diff
@@ -0,0 +1,49 @@
+import logging
+
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import tier1, ManageTest
+from ocs_ci.ocs.node import taint_nodes, get_worker_nodes_not_in_ocs
+from ocs_ci.ocs.resources.pod import get_pod_node, get_plugin_pods
+
+logger = logging.getLogger(__name__)
+
+
+@tier1
+class TestCheckTolerationForCephCsiDriverDs(ManageTest):
+    """
+    Check toleration for Ceph CSI driver DS on non ocs node
+    """
+
+    def test_ceph_csidriver_runs_on_non_ocs_nodes(
+        self, pvc_factory, pod_factory, add_nodes
+    ):
+        """
+        1. Add non ocs nodes
+        2. Taint new nodes with app label
+        3. Check if plugin pods running on new nodes
+        4. Create app-pods on app_nodes
+        """
+
+        # Add worker nodes and tainting it as app_nodes
+        add_nodes(ocs_nodes=False)
+        app_nodes = [node.name for node in get_worker_nodes_not_in_ocs()]
+        logger.info(f"Tainting nodes {app_nodes} ")
+        taint_nodes(taint_label="nodetype=app:NoSchedule", nodes=app_nodes)
```

Comment:
What about to have this done directly in factory? You can have another parameter about tainting label there and you directly know which nodes you created right? So you dont' have to even find them here but directly taint them from your factory.

Reply:
done